### DISCUSSION TITLE
Alerts update behavior clarification

### DISCUSSION THREAD

**Discussion by mikeobr** [weather-gov/api#788] [Q&A]
Hello! I have been trying to understand some of the Update behavior for Alerts via the Cap documentation https://vlab.noaa.gov/web/nws-common-alerting-protocol/cap-documentation#references-1

Is it possible to have the following siuation.
1. Alert 1 is updated by alert 2
2. Alert A is updated by alert B

At this point these are "unique" chains of alerts, but can this situation occur:
3. Alert 1,2,B,A is updated by alert x

I am mostly trying to understand so we can understand an appropriate deduplication strategy.

**Comment by dwhitemv25** [comment]
NWS folks will have to check me on this, but the `expiredReferences` list is exhaustive, and will identify all alerts that have been superceded. As long as you eliminate any alert listed in an expired reference list, you shouldn't have any duplicates.

Your example situation comes up with cancels, where alert X cancels all previous alerts in an event. All of the IDs of any previous alert in that event will be listed in `expiredReferences`. 

That said... humans are involved and weird things happen.
